Cli/Helper/ConfigurationHelper.php

```diff
@@ -90,32 +102,28 @@ private function askValue(InputInterface $input, OutputInterface $output, $varna
             $question = sprintf('What is your %s? ', $varname);
         }
 
-        $validator = function ($v) {
+        $validator = function ($v) use ($varname) {
             if (!$v) {
-                throw new \InvalidArgumentException('Your must answer this question.');
+                throw new \InvalidArgumentException(sprintf('Your must provide a %s!', $varname));
             }
 
             return $v;
         };
 
         $dialog = $this->getHelperSet()->get('dialog');
 
-        return $dialog->askAndValidate($output, $question, $validator, false, $default);
+        return $dialog->askAndValidate($output, $question, $validator, ($input->isInteractive() ? false : 1), $default);
     }
 
     private function saveConfiguration(InputInterface $input, OutputInterface $output, Configuration $configuration)
     {
-        if (!$input->isInteractive()) {
-            return;
-        }
-
         $question = 'Do you want to save this new configuration? [Y/n] ';
         if (PHP_VERSION_ID > 50400) {
             $question = json_encode($configuration->toArray(), JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE)."\n\n".$question;
         }
         $dialog = $this->getHelperSet()->get('dialog');
 
-        if ($dialog->askConfirmation($output, $question)) {
+        if ($dialog->askConfirmation($output, $question) or !$input->isInteractive()) {
```

Comment:
``or`` is a control flow operator, not a boolean operator (which is why it has a much lower precedence). Use ``||``.